File: qb-engineer-server/qb-engineer.core/Models/PaymentListItemModel.cs
namespace QBEngineer.Core.Models;

public record PaymentListItemModel(
    int Id,
    string PaymentNumber,
    int CustomerId,
    string CustomerName,
    string Method,
    decimal Amount,
    decimal AppliedAmount,
    decimal UnappliedAmount,
    DateTimeOffset PaymentDate,
    string? ReferenceNumber,
    DateTimeOffset CreatedAt);

File: qb-engineer-ui/src/app/features/payments/models/payment-list-item.model.ts
export interface PaymentListItem {
  id: number;
  paymentNumber: string;
  customerId: number;
  customerName: string;
  method: string;
  amount: number;
  appliedAmount: number;
  unappliedAmount: number;
  paymentDate: Date;
  referenceNumber: string | null;
  createdAt: Date;
}
